libs/iblprefilter/include/iblprefilter/IBLPrefilterContext.h

```diff
@@ -0,0 +1,196 @@
+     * contains the filter's kernel and sample count.
+     */
+    class SpecularFilter {
+    public:
+        enum class Kernel : uint8_t {
+            D_GGX,        // Trowbridge-reitz distribution
+        };
+
+        /**
+         * Filter configuration.
+         */
+        struct Config {
+            uint16_t sampleCount = 1024;    //!< filter sample count
+            uint8_t levelCount = 5;         //!< number of roughness levels
+            Kernel kernel = Kernel::D_GGX;  //!< filter kernel
+        };
+
+        /**
+         * Filtering options for the current environment.
+         */
+        struct Options {
+            float hdrLinear = 8.0f;      //!< no HDR compression up to this value
+            float hdrMax = 16.0f;        //!< HDR compression between hdrLinear and hdrMax
+            bool generateMipmap = true;  //!< set to false if the environment map already has mipmaps
+        };
+
+        struct ReflectionsOptions {
+            using InternalFormat = filament::Texture::InternalFormat;
+            using Usage = filament::Texture::Usage;
+            uint16_t size = 256;         //!< output environment texture size. Must be Power-of-two.
+            InternalFormat format = InternalFormat::R11F_G11F_B10F; // output environment texture format
+            Usage usage = Usage::NONE;  //!< extra usage added to output environment texture
+        };
+
+        /**
+         * Creates a filter.
+         * @param context IBLPrefilterContext to use
+         * @param config  Configuration of the filter
+         */
+        SpecularFilter(IBLPrefilterContext& context, Config config);
+
+        /**
+         * Creates a filter with the default configuration.
+         * @param context IBLPrefilterContext to use
+         */
+        explicit SpecularFilter(IBLPrefilterContext& context);
+
+        /**
+         * Destroys all GPU resources created during initialization.
+         */
+        ~SpecularFilter() noexcept;
+
+        SpecularFilter(SpecularFilter const&) = delete;
+        SpecularFilter& operator=(SpecularFilter const&) = delete;
+        SpecularFilter(SpecularFilter&& rhs) noexcept;
+        SpecularFilter& operator=(SpecularFilter&& rhs);
+
+        /**
+         * Creates a texture suitable for being used as a reflection map
+         * @param options options for the texture creation
+         * @return a Texture object owned by the caller.
+         */
+        filament::Texture* createReflectionsTexture(ReflectionsOptions options);
+
+        /**
+         * Generates a prefiltered cubemap.
+         * @param options                   Options for this environment
+         * @param environmentCubemap        Environment cubemap (input). Can't be null.
+         * @param outReflectionsTexture     Output prefiltered texture or if null it is created.
+         * @return returns outReflectionsTexture
+         */
+        filament::Texture* operator()(Options options,
```

Comment:
I think it's a bit unexpected to make this API a function call operator, when we also have the `createReflectionsTexture` method. I'd probably make this into a method called `createPrefilteredCubemapTexture` or something.

Reply:
Actually, I think my question is rather: why is `createReflectionsTexture` public?

Reply:
I wanted to avoid "createXXX" because it's more than that -- it's launch the computation. I didn't want to call it `.filter()`. I though of `.process()`. But then I thought in fact it doesn't need a name. It's just the "do it" method.

Reply:
`createReflectionsTexture` is more like a helper, it'll create the texture with the right `Usage` for instance. But you don't have to use it.